File: src/main/java/alfheim/common/network/MessageHotSpellC.java
package alfheim.common.network;

import cpw.mods.fml.common.network.simpleimpl.IMessage;
import io.netty.buffer.ByteBuf;

public class MessageHotSpellC implements IMessage {

	public int[] ids;
	
	public MessageHotSpellC() {
		ids = new int[12];
	}
	
	public MessageHotSpellC(int[] ids) {
		this.ids = ids.clone();
	}
	
	@Override
	public void toBytes(ByteBuf buf) {
		for (int id : ids) buf.writeInt(id);
	}
	
	@Override
	public void fromBytes(ByteBuf buf) {
		for (int i = 0; i < ids.length; i++) ids[i] = buf.readInt();
	}
}

File: src/main/java/alfheim/common/network/MessageHotSpellC.kt
package alfheim.common.network

import alfheim.client.core.handler.PacketHandlerClient
import cpw.mods.fml.common.network.simpleimpl.IMessage
import cpw.mods.fml.common.network.simpleimpl.IMessageHandler
import cpw.mods.fml.common.network.simpleimpl.MessageContext
import io.netty.buffer.ByteBuf

class MessageHotSpellC: IMessage {
	
	val ids: IntArray
	
	constructor() {
		ids = IntArray(12)
	}
	
	constructor(ids: IntArray) {
		this.ids = ids.clone()
	}
	
	override fun toBytes(buf: ByteBuf) {
		for (id in ids) buf.writeInt(id)
	}
	
	override fun fromBytes(buf: ByteBuf) {
		for (i in ids.indices) ids[i] = buf.readInt()
	}
	
	class Handler: IMessageHandler<MessageHotSpellC, IMessage> {
		
		override fun onMessage(packet: MessageHotSpellC, message: MessageContext): IMessage? {
			PacketHandlerClient.handle(packet)
			return null
		}
	}
}
